config/routes.rb

```diff
@@ -15,18 +15,22 @@
 
   namespace 'jira' do
     namespace 'status' do
+      # These require :service_name as a query string param
       resources :push, only: [:edit, :update]
     end
   end
 
-  get 'sha/:id' => 'jira/status/push#edit'
-  get 'deploy_email/:id' => 'jira/status/push#deploy_email'
-  get 'branch/:branch' => 'jira/status/push#branch'
-  get 'summary' => 'jira/status/push#summary'
+  # Push routes
+  get  ':service_name/sha/:id'          => 'jira/status/push#edit'
+  get  ':service_name/deploy_email/:id' => 'jira/status/push#deploy_email'
+
+  get  'branch/:branch' => 'jira/status/push#branch'
+  get  'summary'        => 'jira/status/push#summary'
```

Comment:
Wouldn't we also need a summary per `service_name`?

Reply:
updated to have service specific summary pages